pyomo/contrib/doe/doe.py

```diff
@@ -964,19 +1001,22 @@ def initialize_fim(m, j, d):
                 initialize=identity_matrix,
             )
 
-        # move the L matrix initial point to a dictionary
-        if type(self.L_initial) != type(None):
-            dict_cho = {}
-            for i, bu in enumerate(model.regression_parameters):
-                for j, un in enumerate(model.regression_parameters):
-                    dict_cho[(bu, un)] = self.L_initial[i][j]
+        # if cholesky, define L elements as variables
+        if self.Cholesky_option and self.objective_option == ObjectiveLib.det:
 
-        # use the L dictionary to initialize L matrix
-        def init_cho(m, i, j):
-            return dict_cho[(i, j)]
+            # move the L matrix initial point to a dictionary
+            if type(self.L_initial) != type(None):
+                dict_cho = {}
+                # Loop over rows
+                for i, bu in enumerate(model.regression_parameters):
+                    # Loop over columns
+                    for j, un in enumerate(model.regression_parameters):
+                        dict_cho[(bu, un)] = self.L_initial[i][j]
```

Comment:
For what it's worth, dictionary comprehensions are usually faster:
```python
dict_cho = {
    (bu, un): self.L_initial[i][j] 
    for i, bu in enumerate(model.regression_parameters)
    for j, un in enumerate(model.regression_parameters)
}
```

Reply:
Fixed, thank you.